File: app/src/main/java/de/tum/in/tumcampus/models/Person.java
package de.tum.in.tumcampus.models;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

import java.io.Serializable;

/**
 * A person, often an {@link Employee} working at TUM.
 *
 * Note: This model is based on the TUMOnline web service response format for a
 * corresponding request.
 */
@Root(name = "row", strict = false)
public class Person implements Serializable {

	public static final String FEMALE = "W";

	public static final String MALE = "M";
	private static final long serialVersionUID = -5210814076506102292L;

	@Element(name = "geschlecht", required = false)
	private String gender;

	@Element(name = "obfuscated_id")
	private String id;

	@Element(name = "vorname")
	private String name;

	@Element(name = "familienname")
	private String surname;

	public String getGender() {
		return gender;
	}

	public String getId() {
		return id;
	}

	public String getName() {
		if (name == null) {
			return "";
		}
		return name;
	}

	public String getSurname() {
		if (surname == null) {
			return "";
		}
		return surname;
	}

	public void setId(String id) {
		this.id = id;
	}

	public void setName(String name) {
		this.name = name;
	}

    public void setSurname(String surname) {
        this.surname = surname;
    }

    @Override
    public String toString() {
        return getName() + " " + getSurname();
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/person/model/Person.kt
package de.tum.`in`.tumcampusapp.component.tumui.person.model

import com.google.gson.Gson
import com.tickaroo.tikxml.annotation.PropertyElement
import com.tickaroo.tikxml.annotation.Xml
import de.tum.`in`.tumcampusapp.component.other.general.RecentsDao
import de.tum.`in`.tumcampusapp.component.other.general.model.Recent
import java.io.Serializable

/**
 * A person, often an [Employee] working at TUM.
 *
 *
 * Note: This model is based on the TUMOnline web service response format for a
 * corresponding request.
 */
@Xml(name = "row")
data class Person(
    @PropertyElement(name = "geschlecht")
    var gender: String = "",
    @PropertyElement(name = "obfuscated_id")
    var id: String = "",
    @PropertyElement(name = "vorname")
    var name: String = "",
    @PropertyElement(name = "familienname")
    var surname: String = "",
    @PropertyElement(name = "bild_url")
    var imageUrl: String = "",
) : Serializable {

    fun getFullName() = "$name $surname"

    fun getFullImageUrl() =
            "https://campus.tum.de/tumonline/$imageUrl".replace("&amp;", "&")

    companion object {

        private const val serialVersionUID = -5210814076506102292L

        const val FEMALE = "W"
        const val MALE = "M"

        @JvmStatic fun toRecent(person: Person) : Recent {
            val gson = Gson()
            val jsonString = gson.toJson(person)
            return Recent(name = jsonString, type = RecentsDao.PERSONS)
        }

        @JvmStatic fun fromRecent(r: Recent): Person {
            val gson = Gson()
            return gson.fromJson(r.name, Person::class.java)
        }
    }
}
